test/e2e/components/vl-upload.js

```diff
@@ -1,6 +1,86 @@
 const { VlElement } = require('vl-ui-core').Test;
+const { By } = require('selenium-webdriver');
 
-class VlUpload extends VlElement {  
+class VlUpload extends VlElement { 
+	async uploadFile(path) {
+		const input = await this.shadowRoot.findElement(By.css('input[type="file"].dz-hidden-input'));
+		return input.sendKeys(path);
+	}
+	
+	async getFiles() {
+		const files = await this.shadowRoot.findElements(By.css(".vl-upload__files__container .vl-upload__file"));
+		return await Promise.all(files.map(file => new VlUploadFile(this.driver, file)));
+	}
+	
+	async isError() {
+		return this.hasAttribute("error");
+	}
+	
+	async isFullBodyDrop() {
+		return this.hasAttribute("full-body-drop");
+	}
+	
+	async removeFiles() {
+		const files = await this.getFiles();
+		await Promise.all(files.map(f => f.remove()));
+	}
+	
+	async getMaximumFilesize() {
+		return this.getAttribute("max-size");
+	}
+
+	async getMaximumNumberOfAllowedFiles() {
+		return this.getAttribute("max-files");
+	}
+
+	async getAcceptedFileTypes() {
+		return this.getAttribute("accepted-files");
+	}
+	
+	async isDuplicatesDisallowed() {
+		const duplicatesDisallowed = await this.getAttribute("disallow-duplicates");
+		return duplicatesDisallowed == "true";
+	}
+}
+
+class VlUploadFile extends VlElement  {
+	
+	async getName() {
+		const nameSpan = await this.findElement(By.css("span[data-dz-name]"));
+		return nameSpan.getText();
+	}
+
+	async getSize() {
+		const sizeSpan = await this.findElement(By.css("span[data-dz-size]"));
+		return sizeSpan.getText();
+	}
+	
+	async remove() {
+		const removeButton = await this.findElement(By.css("button.vl-upload__file__close"));
```

Comment:
Je had zeker miserie als de upload nog bezig was en hij een alert opgooit om te bevestigen dat je de upload wilt annuleren?
Moeten we dat ook niet implementeren, indien mensen dat willen testen?

Reply:
goh, er is niks van progress bar of zo te zien, dus ik had dat voorlopig niet gedaan ...
wat ik kan doen daar is mijn stub file server niks laten terugsturen, dan moet ge het annuleren met nen alert denk ik. maar ik denk ook niet dat ze dat weten bij webuniversum want da lijkt me niet zo "huissyle"-ish ...

Reply:
maar ik kan er dus wel een test voor schrijven. tis niet omdat het bij webuniversum niet goed opgelost is, dat ik het niet kan testen.

Reply:
Ja klopt, lijkt me inderdaad vreemd... 
@coemans?

Reply:
bwa, ik schrijf de test wel gewoon, da zal nu ook geen uren werk zijn.

Reply:
![maxresdefault](https://user-images.githubusercontent.com/254712/76081754-10d9a700-5faa-11ea-88c4-631569894e83.jpg)